### PR TITLE
Betrue/fix grouped list headers

### LINKED ISSUE #2076: GroupedList: Various fit and finish issues
### Bug Report

Are you willing to submit a PR to fix? (Yes)
Requested priority: (Normal) 

#### Describe the issue:

There are various issues with the current group item for a grouped list:

1. Expand arrow is too small
2. There is no space between the group name and count
3. The height is too tall, should be the same as a file
4. Margin is too close to left of selection circle

image

### PR DESCRIPTION
**Pull request by betrue-final-final**
#### Pull request checklist

- [x] Addresses an existing issue: Fixes #2076
- [x] Include a change request file using `$ npm run change`

1. Fixed spacing and vertical position of expand arrow
2. Added space between title and count
3. Changed to use flex box
4. Updated interactive state colors to use proper semantic slots

### CODE REVIEW

**Comment by betrue-final-final** [comment]
Here's a screen shot. Note that the selection circle can't be resized until we get a thick enough checkmark so it doesn't fall apart visually.
image

File: packages/office-ui-fabric-react/src/components/GroupedList/GroupHeader.scss [hunk 1]
```diff
@@ -10,35 +17,33 @@ $groupHeader-ease-in-back: cubic-bezier(0.600, -0.280, 0.735, 0.045);
...
-    background: $ms-color-themeLighter;
+    background: $detailsList-item-selected-color;
   }
   &.rootIsSelected:hover {
-    background: $ms-color-themeLight;
+    background: $detailsList-item-selected-hover-color;
   }
 }
 
+.focusZone {
```
Comment (aditima) [inline packages/office-ui-fabric-react/src/components/GroupedList/GroupHeader.scss]: Should we call this focusZone or something more meaningful like say groupHeaderContainer?

Comment (betrue-final-final) [inline packages/office-ui-fabric-react/src/components/GroupedList/GroupHeader.scss]: I can change that. The reason we called it that was that was the name in the DOM...I think that's what that was?

**Comment by betrue-final-final** [comment]
Changed class name to GroupHeaderContainer

File: packages/office-ui-fabric-react/src/components/GroupedList/GroupHeader.scss [hunk 2]
```diff
@@ -52,15 +57,14 @@ $groupHeader-ease-in-back: cubic-bezier(0.600, -0.280, 0.735, 0.045);
 
 .title {
   @include ms-font-xl;
-  padding: 14px 6px;
-  display: inline-block;
   cursor: pointer;
   outline: 0;
+  padding-left: 6px;
```
Comment (dzearing) [inline packages/office-ui-fabric-react/src/components/GroupedList/GroupHeader.scss]: Nope. Can't do this.

File: packages/office-ui-fabric-react/src/components/GroupedList/GroupHeader.scss [hunk 3]
```diff
@@ -52,15 +57,14 @@ $groupHeader-ease-in-back: cubic-bezier(0.600, -0.280, 0.735, 0.045);
...
 }
 
 .expand {
   width: 36px;
-  height: 40px;
+  padding-top: 4px;
```
Comment (dzearing) [inline packages/office-ui-fabric-react/src/components/GroupedList/GroupHeader.scss]: Why is this necessary? Are you trying to center something? If so this is very wrong.

Comment (betrue-final-final) [inline packages/office-ui-fabric-react/src/components/GroupedList/GroupHeader.scss]: The chevron icon isn't vertically centered, and it needs to align with the baseline and lower x-height of the title. Otherwise it's misaligned.

Comment (betrue-final-final) [inline packages/office-ui-fabric-react/src/components/GroupedList/GroupHeader.scss]: Good point on the RTL for asymmetric horizontal padding… I think it can do fine without that. However I don’t recommend we get rid of the top padding. The world is not perfectly mathematical when it comes to visuals, and without it, the code may be slightly more elegant, but the product will be wonky. See this example:
image


As you can see, without top padding, the chevron looks misaligned towards the top, like it has a lump under its bed. There are a few things going on here.

- The text is optically centered based on the lower x-height, and not the capital or entire size of the text control.

- The chevron visually weights away from the point, and the glyph itself is set higher in its canvas. Both of these things make the chevron even appear higher than the circle check that it’s supposed to be aligned with. This is the only icon we will use here.

If I remember correctly, top padding has no bearing on RTL, so I really think we can have this one line of padding so our product doesn’t look weird. I’m not too worried about this causing any issues with future work, as it’s a small isolated case that we need to deal with these typographic considerations.

Here's a closer look, with the top being without top padding.
image

File: packages/office-ui-fabric-react/src/components/GroupedList/GroupHeader.tsx [hunk 4]
```diff
@@ -109,7 +109,7 @@ export class GroupHeader extends BaseComponent<IGroupDividerProps, IGroupHeaderS
               // so far. That's the reasons we need to use "+" to show we might have more items than count
               // indicates.
             }
-            <span>({ group.count }{ group.hasMoreData && '+' })</span>
+            <span>{ ' ' }({ group.count }{ group.hasMoreData && '+' })</span>
```
Comment (dzearing) [inline packages/office-ui-fabric-react/src/components/GroupedList/GroupHeader.tsx]: The extra space here is not RTL friendly. Instead, just put a class on this, and set the padding to `0 8px`. That will enforce that there is consistent padding between the name and the count.

**Comment by betrue-final-final** [comment]
Updated spacing between title and count to use symmetrical CSS instead of a space in the TSX file.
